theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
I don't understand the second option. The part that I don't understand is "The other implementation would be to take inputs of shape (batches, in_channels, ih, iw) and kernel of shape (num_filters, in_channels // num_groups , kh, kw)"

There, how the inputs can be of size "in_channels // num_groups"? I don't get that.